README.md

```diff
@@ -1040,20 +1040,7 @@ Doesn't support static type inference. 😕
 
 Yup is a full-featured library that was implemented first in vanilla JS, with TypeScript typings added later.
 
-Yup supports static type inference, but unfortunately the inferred types aren't actually correct. Currently, the yup package treats all object properties as optional by default:
-
-```ts
-const schema = yup.object({
-  asdf: yup.string(),
-});
-schema.validate({}); // passes
-
-type SchemaType = yup.InferType<typeof schema>;
-// returns { asdf: string }
-// should be { asdf?: string }
```

Comment:
After re-reading this I'm not entirely sure the original statement made sense, so I'm not sure what you were saying initially.

It does indeed return `{asdf: string}` and if you specify `nonRequired()`, it returns `{asdf?: string}`. In the paragraph above, it says "Currently, the yup package treats all object properties as optional by default" which is the inverse of what's true as far as I can tell.

I don't know that I see the current default behavior of yup as a concern.

Here's their explanation https://github.com/jquense/yup#why-does-infertype-not-default-to-nonrequired

Reply:
Okay, I'm sorry, this took a little time to digest and I'm with you now. You're saying that the library itself considers them optional, but that the type marks them as non-optional. They don't totally explain their reasoning in the link I provided above. In the end, I think this is a valid criticism as it's going to be surprising to some.

I'll go ahead and close this out, sorry for the noise!